### PR TITLE
Add a health check endpoint on the search server.

### LINKED ISSUE #268: Add a healthcheck endpoint on search server
This is useful when you want to host search servers behind a load balancer. The load balancer generally expects an endpoint to check if the server is correctly running.

I think we did that for barrel.

### PR DESCRIPTION
**Pull request by mosuka**
### Context / purpose
Add a health check endpoint on the search server. https://github.com/quickwit-inc/quickwit/issues/268

### Description
Port the health check functionality from Barrel.

### How was this PR tested?

```
$ cargo run -- serve --host=0.0.0.0 --index-uri=file:///home/minoru/tmp/quickwit_data/indices/hdfslogs-idx --port=8080
```

```
$ curl 'http://0.0.0.0:8080/health/livez'
```

### Checklist
- [x] tested locally
- [x] added unit tests
- [x] included documentation

Close #268

### CODE REVIEW

**Comment by fmassot** [comment]
Thanks @mosuka for the PR. But I can't merge that for the moment because:
- I think we don't need HTML but only a json response
- We can start by something very simple, like a 200 empty response is good 
- We need to define what health checks we want before

**Comment by mosuka** [comment]
@fmassot 
I had mistakenly thought that it ported from barrel.
I’ll define the health check specification.
Thanks.

**Comment by fmassot** [comment]
Can you remove this file?

**Comment by mosuka** [comment]
@fmassot 
I have simplified the code by removing unnecessary code.
Please take another look.

File: quickwit-serve/src/rest.rs [hunk 1]
```diff
@@ -528,6 +547,17 @@ mod tests {
...
     }
 
+    #[tokio::test]
+    async fn test_rest_search_api_health_check_livez() {
+        let rest_search_api_filter = liveness_check_handler();
+        let resp = warp::test::request()
+            .path("/health/livez")
+            .reply(&rest_search_api_filter)
+            .await;
+        println!("{:?}", resp);
```
Comment (fmassot) [inline quickwit-serve/src/rest.rs:557]: remove the print

**Review by fmassot — CHANGES_REQUESTED** [review]

File: quickwit-serve/src/rest.rs (hunk 1 above)
Comment (mosuka) [reply quickwit-serve/src/rest.rs]: Removed it. Thanks.

**Review by fmassot — APPROVED** [review]

**Pull request merged**